### ISSUE TITLE
Treat floating module doctring like assigning to `__doc__`

### ISSUE THREAD

**Issue by Avasam** [microsoft/pyright#9404]
If you have a question about a behavior that you’re seeing in pyright, consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Is your feature request related to a problem? Please describe.**
Trying to use a module's doctring leads to false-positives with `None`. This doesn't happen if explicitly assigning to `__doc__`, but that doesn't feel very pythonic.

Example from https://github.com/microsoft/python-type-stubs/blob/main/utils/count_ids.py
```py
"""Count IDs.

Usage:
  count_ids [--path=<root>] [--suffix=<filesuffix>] [--pat=<pat>] [--uniq]
  count_ids -h | --help
  count_ids --version

Options:
  -h --help               Show this screen.
  --version               Show version.
  --path=<root>           Directory to scan (default is current working directory)
  --suffix=<filesuffix>   File name suffix to restrict to (default is all files)
  --pat=<pat>             A regular expression to use to extract IDs
  --uniq                  Only output IDs that occur once only (with their context)
"""

import docopt

# ...

if __name__ == "__main__":
    # Argument of type "str | None" cannot be assigned to parameter "docstring" of type "str" in function "docopt"
    #   Type "str | None" is not assignable to type "str"
    #     "None" is not assignable to "str"Pylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)
    args = docopt.docopt(__doc__, version="Count IDs 0.1")
    # ...
```

**Describe the solution you’d like**
Doing the following works and makes `__doc__` a literal string
```py
__doc__ = """Count IDs.

Usage:
  count_ids [--path=<root>] [--suffix=<filesuffix>] [--pat=<pat>] [--uniq]
  count_ids -h | --help
  count_ids --version

Options:
  -h --help               Show this screen.
  --version               Show version.
  --path=<root>           Directory to scan (default is current working directory)
  --suffix=<filesuffix>   File name suffix to restrict to (default is all files)
  --pat=<pat>             A regular expression to use to extract IDs
  --uniq                  Only output IDs that occur once only (with their context)
"""
```

But I'd like the example given above to work without having to explicitly assign to `__doc__`

**Comment by erictraut** [comment]
Pyright is working as intended here. This topic has been discussed previously. In fact, pyright at one point did what you're suggesting, but I undid the change based on feedback.

A module's `__doc__` attribute is writable, so it's not type safe to assume that the presence of a docstring at the top of a module (or a class or a function) narrows the type to `None`. 

You can add a `assert __doc__ is not None` to eliminate this error.
